logic/nodes.go

```diff
@@ -75,6 +75,16 @@ func UpdateNodeCheckin(node *models.Node) error {
 	return database.Insert(node.ID.String(), string(data), database.NODES_TABLE_NAME)
 }
 
+// UpsertNode - updates node in the DB
+func UpsertNode(newNode *models.Node) error {
```

Comment:
Use the existing `UpdateNode`?

Reply:
it's not the ideal func to use for this purpose